pkg/controller/portworxdiag/pod.go

```diff
@@ -0,0 +1,144 @@
+package portworxdiag
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/libopenstorage/operator/drivers/storage/portworx"
+
+	pxutil "github.com/libopenstorage/operator/drivers/storage/portworx/util"
+	corev1 "github.com/libopenstorage/operator/pkg/apis/core/v1"
+	portworxv1 "github.com/libopenstorage/operator/pkg/apis/portworx/v1"
+	"github.com/libopenstorage/operator/pkg/util"
+	k8sutil "github.com/libopenstorage/operator/pkg/util/k8s"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	PortworxDiagLabel = "portworx-diag"
+)
+
+func v1Volume(name, path string, hpType v1.HostPathType) v1.Volume {
+	return v1.Volume{
+		Name: name,
+		VolumeSource: v1.VolumeSource{
+			HostPath: &v1.HostPathVolumeSource{
+				Path: path,
+				Type: &hpType,
+			},
+		},
+	}
+}
+
+func volumes() []v1.Volume {
+	return []v1.Volume{
+		v1Volume("dockersock", "/var/run/docker.sock", v1.HostPathUnset),
+		v1Volume("containerddir", "/run/containerd", v1.HostPathUnset),
+		v1Volume("criosock", "/var/run/crio", v1.HostPathUnset),
+		v1Volume("crioconf", "/etc/crictl.yaml", v1.HostPathFileOrCreate),
+		v1Volume("optpwx", "/opt/pwx", v1.HostPathUnset),
+		v1Volume("procmount", "/proc", v1.HostPathUnset),
+		v1Volume("sysdmount", "/etc/systemd/system", v1.HostPathUnset),
+		v1Volume("dbusmount", "/var/run/dbus", v1.HostPathUnset),
+		v1Volume("varlibosd", "/var/lib/osd", v1.HostPathUnset),
+		v1Volume("diagsdump", "/var/cores", v1.HostPathUnset),
+		v1Volume("etcpwx", "/etc/pwx", v1.HostPathUnset),
+		v1Volume("journalmount1", "/var/run/log", v1.HostPathUnset),
+		v1Volume("journalmount2", "/var/log", v1.HostPathUnset),
+		v1Volume("registration-dir", "/var/lib/kubelet/plugins_registry", v1.HostPathDirectoryOrCreate),
+		v1Volume("csi-driver-path", "/var/lib/kubelet/plugins/pxd.portworx.com", v1.HostPathDirectoryOrCreate),
```

Comment:
this kubelet path here can change based on where we're running - 
https://github.com/libopenstorage/operator/blob/4002f44a87d6ee54765b1a883eae7d5e1e08e175/drivers/storage/portworx/deployment.go#L1475C30-L1475C30

I'm not sure if that matters here, but just FYI

Reply:
I think we're hoping to remove this volume/mount entirely so I'll leave it for now but good reminder, thank you!